### DISCUSSION TITLE
Gravatar profile picture not displayed on GitHub

### DISCUSSION THREAD

**Discussion by AlbertoFabbri93** [community/community#53616] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Hi,

I have uploaded a profile picture on Gravatar, and on GitHub I am currently using the email address associated with my Gravatar profile as my primary email address, but the picture is not being picked up by GitHub.

Are Gravatar profiles still supported by GitHub?

**Comment by NJacksonDev** [comment]
Yes, they are. Please see a recent/relevant discussion below re: changing gravatar profile pictures. 

As noted in the discussion, please note that GitHub will use your G-rated avatar (not PG and above), so be sure to set that.

https://stackoverflow.com/questions/10221280/change-avatar-next-to-github-username-for-commits

**Reply by AlbertoFabbri93** [reply]
Thank you very much for the quick reply, I checked and my avatar is rated correctly.

I also read the stackoverflow thread you linked but I was unable to solve the problem. Nevertheless it got me thinking that the problem may be related to the fact that I used to have a different email address and all the commits on my "personal" repositories on GitHub are signed with that old address.
Now, I am using my new email address (the one connected to Gravatar) as part of the work that I am doing on an organization here in GitHub.
Can this be the problem? If yes, how do I solve it?

**Reply by andrey-helldar** [reply]
I have the same problem. After I changed my primary email address, Gravatar in my profile stopped working.

**Comment by ghost** [comment]
Not an answer, but I'm having the same problem... My email for Gravatar and GitHub match but it just doesn't get my Gravatar ID, even in the API: 

```js
{
  "login": "BetaPictoris",
  "id": 65696362,
  "node_id": "MDQ6VXNlcjY1Njk2MzYy",
  "avatar_url": "https://avatars.githubusercontent.com/u/65696362?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/BetaPictoris",
  // ...
}
```

**Reply by mariaparkr** [reply]
Gravatar profile pictures are not displayed on GitHub by default.
GitHub uses a separate avatar system, and you need to upload your profile picture directly on GitHub.
Gravatar profile pictures are primarily associated with email addresses and are not automatically synced with GitHub accounts.
For more Detail click here:   https://toysshopforkids.com

**Reply by ghost** [reply]
Hey @mariaparkr, I think you put the wrong link.... but also Gravatar used to work.

**Reply by Grohden** [reply]
gravatar should work, its in their docs
https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/personalizing-your-profile#resetting-your-profile-picture-to-the-identicon


I also don't know why its not working for me.. I have multiple emails but my primary one is the one in which I use my gravatar

**Reply by Grohden** [reply]
https://developer.github.com/changes/2014-09-05-removing-gravatar-id/ @BetaPictoris about the gravatar_id being empty

**Reply by ghost** [reply]
Ah, okay, Gravatars still don't work, I've removed and readded my custom profile picture a couple of times, and also removed my secondary email, so that my primary one (and the one linked to my Gravatar account) is on top, I saw in a Stack Overflow forum that this might help - it didn't.

**Comment by DavidAntliff** [comment]
Gravatar on GitHub stopped working for me months ago, thought it was just me...

**Comment by upward4** [comment]
I'm also having this issue with a GitHub account. I've checked my gravatar.com's profile settings and all seems fine (G-rated, privacy set to public, etc.). I'm able to retrieve the picture by constructing its URL with the hashed email, i.e. `https://www.gravatar.com/avatar/<email_hash>?s=200`.

This seems like a bug on GitHub's end.

**Comment by bradgignac** [comment]
I'm seeing the same. I used Gravatar with GitHub successfully for years, but it broke when I updated email addresses after a job change. Everything _appears_ to be set up correctly on both Gravatar and GitHub, but it still isn't working.

**Comment by sinhote** [comment]
I just wanted to comment that I am facing the same issue. I cannot see my Gravatar image (G rated, public, working on other sites, etc.) in Github.

**Reply by xyz327** [reply]
it's same for me. G rated.  working for other site, but not working in github..

**Comment by daleswift** [comment]
Not working for me either. Did work once, but stopped sometime, not sure when. No changes - just stopped working

**Comment by andrey-helldar** [comment]
Gravatar does not connect to my GitHub profile. No matter how much time has passed - a stub is constantly displayed.

![image](https://github.com/community/community/assets/10347617/91d062d9-54c7-4645-a892-99b687226baf)

For example, here's an image of my gravatar (https://www.gravatar.com/avatar/ee1fe7d89b09d3025ab9ce47aaf55ed8?s=100):

![gravatar](https://www.gravatar.com/avatar/ee1fe7d89b09d3025ab9ce47aaf55ed8?s=100)

At the moment my avatar is uploaded through the file directly (https://avatars.githubusercontent.com/u/10347617?v=4)

But I have to say that this behavior appeared after **I changed my email address**. It happened in 2021.

**Comment by nithin-mk** [comment]
This might solve the issue: https://webapps.stackexchange.com/a/28160/372743
